doc/user_guide/configuration/security.rst

```diff
@@ -50,17 +74,138 @@ Specific examples
+- Exasol SAAS running in the cloud uses proper certificate generated by public certificate authority. It does not require any extra setup.
+- Exasol Docker uses self-signed SSL certificate by default. You may generate a proper SSL certificate and submit it for use via the ConfD API. More details are available on:
+
+   - `GitHub for Exasol Docker <https://github.com/exasol/docker-db>`_
+   - `ConfD API <https://docs.exasol.com/db/latest/confd/confd.htm>`_
+   - `confd_client cert_update <https://docs.exasol.com/db/latest/confd/jobs/cert_update.htm>`_
+
+Setup
+^^^^^
+
+In order to validate a certificate which will be provided in your PyExasol connection
+(see :ref:`Certificate Handling in PyExasol <certificate_in_pyexasol>`),
+you will need to have the certificate setup for your PyExasol usage: either on your
+:ref:`client_machine` or :ref:`inside_a_udf`.
+
+.. _client_machine:
+
+Client machine
+""""""""""""""
+
+#. Public CA
+    * The certificate should already be in the operating system truststore of the client machine.
+#. Private CA (Corporate CA)
+    * Your IT should add it to operating system truststore of the client machine.
+#. Self-signed Certificate
+    * Your IT should add it to operating system truststore of the client machine.
+        1. DBA needs to fetch the certificate from the Exasol Cluster.
+        2. Client Machine Admin needs to add it to the  operating system truststore.
+    * Or, in case of a unprivileged user and the user can access the certificate of the Exasol database you can specify the certificate during connect.
+    * For testing with a local DB you can disable the certificate validation (however, this should **NEVER** be used for production).
+
+.. _inside_a_udf:
+
+Inside a UDF
+""""""""""""
+
+#. Public CA
+    * The certificate should already be in the operating system truststore of the client machine.
+#. Private CA (Corporate CA)
+    * Your DBA should upload the certificate to BucketFS and you should pass it to the connect inside of the UDF.
+        * Note: The operating system truststore is part of the SLC and can only be changed during SLC creation.
+          While you run a UDF, the operating system truststore is read-only.
+#. Self-signed Certificate
+    * Your DBA or you should upload the certificate to BucketFS and you should pass it to the connect inside of the UDF.
+        * Note: The operating system truststore is part of the SLC and can only be changed during SLC creation.
+          While you run a UDF, the operating system truststore is read-only.
+    * For testing with a local DB you can disable the certificate validation (however, this should **NEVER** be used for production).
+
+.. _certificate_in_pyexasol:
+
+Handling in PyExasol
+^^^^^^^^^^^^^^^^^^^^
+
+Passing into the Connection
+"""""""""""""""""""""""""""
+
+This is how an unprivileged user can specify the certificate when making the connection.
```

Comment:
also want to change how that this is how it would work for UDFs, right? passing the cert.